include/boost/geometry/algorithms/FrechetDistance.hpp

```diff
@@ -0,0 +1,176 @@
+/*  
+        Boost (Gsoc '18)
+        SimmilarityBetweenGeometries
+        Author : Yaghyavardhan Sing Khangarot
+        Institute: IIIT Hyderabad
+*/
+
+#include <algorithm> 
+
+#ifdef BOOST_GEOMETRY_DEBUG_FRECHET_DISTANCE
+#include <iostream>
+#endif
+
+#include <iterator>
+#include <utility>
+#include <vector>
+#include <limits>
+
+#include <boost/geometry/geometry.hpp>
+#include <boost/geometry/geometries/linestring.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <boost/range.hpp>
+#include <boost/mpl/assert.hpp>
+#include <boost/geometry/core/tag.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace frechet_distance
+{
+
+template <typename size_type,typename result_type>
+class coup_mat
+{
+public:
+    coup_mat(size_type w, size_type h)
+        : m_data(w * h,-1), m_width(w), m_height(h)
+    {}
+
+    result_type & operator()(size_type i, size_type j)
+    {
+        BOOST_ASSERT(i < m_width && j < m_height);
+        return m_data[j * m_width + i];
+    }
+
+private:
+    std::vector<result_type> m_data;
+    size_type m_width;
+    size_type m_height;
+};
+
+struct linestring_linestring
+{
+    template <typename Linestring1, typename Linestring2, typename Strategy>
+    static inline typename distance_result
+        <
+            typename point_type<Linestring1>::type,
+            typename point_type<Linestring2>::type,
+            Strategy
+        >::type apply(Linestring1 const& ls1, Linestring2 const& ls2, Strategy const& strategy)
+    {   
+        typedef typename distance_result
+            <
+                typename point_type<Linestring1>::type,
+                typename point_type<Linestring2>::type,
+                Strategy
+            >::type result_type;
+        typedef typename boost::range_size<Linestring1>::type size_type;
+        
+
+        boost::geometry::detail::throw_on_empty_input(ls1);
+        boost::geometry::detail::throw_on_empty_input(ls2);
+
+        size_type const a = boost::size(ls1);
+        size_type const b = boost::size(ls2);
```

Comment:
`size_type` of `Linestring1` could be different than `size_type` of `Linestring2`.

Reply:
Right but you have to use something in coup_map. You could use the greater type or leave it as it is for now. In practice this should be ok.

Reply:
I have updated the code and defined size_type1 for Linestring1 and size_type2 for Linestring2. And also used both these size_types in creating coup_mat.